### ISSUE TITLE
Better document StyleGuide's initialization options

### ISSUE THREAD

**Issue by nchammas** [PyCQA/pycodestyle#463]
It's not clear to the user using pep8 as a library how to replicate pep8's command-line behavior of loading settings from config files like tox.ini.

The doc on [advanced usage](https://github.com/PyCQA/pep8/blob/4dc42d842274ba27d2724e76eb83ff69e7db226f/docs/advanced.rst#automated-tests) gives a good start to using pep8 as a library, but more details are hard to come by.

`help(pep8.StyleGuide)` is also unhelpful:

```
class StyleGuide(builtins.object)
 |  Initialize a PEP-8 instance with few options.
 |  
 |  Methods defined here:
 |  
 |  __init__(self, *args, **kwargs)
 |      Initialize self.  See help(type(self)) for accurate signature.
```

I can't get any variation of `help(pep8.StyleGuide)` or `help(type(...))` to give me an "accurate signature". I think that's because the class doesn't actually offer one.

A few questions:

1. What is the correct way to initialize `StyleGuide` so that pep8 loads settings from tox.ini or other config files?
2. Is there any reason `StyleGuide`'s [initialization signature](https://github.com/PyCQA/pep8/blob/4dc42d842274ba27d2724e76eb83ff69e7db226f/pep8.py#L1788-L1795) shouldn't be updated to something more documentation-friendly?

  For example:

  ```python
def __init__(
        self,
        checker_class=Checker,
        parse_argv=False,
        config_file=False,
        parser=None,
        *args, **kwargs)
  ```
3. Would it be appropriate to add some additional detail to the [advanced usage docs](https://github.com/PyCQA/pep8/blob/4dc42d842274ba27d2724e76eb83ff69e7db226f/docs/advanced.rst) on how to load settings from config files? Users making the transition from command-line to library use will typically be looking for that behavior.

**Comment by nchammas** [comment]
I just realized that [this line](https://github.com/PyCQA/pep8/blob/4dc42d842274ba27d2724e76eb83ff69e7db226f/pep8.py#L1792) should probably read:

```python
config_file = kwargs.pop('config_file', None)
```

It's functionally equivalent to the current line, but better communicates that `config_file` is a thing and not a flag.

**Comment by nchammas** [comment]
OK, so I figured out that the invocation I'm looking for is this:

```python
pep8.StyleGuide(config_file='tox.ini')
```

That answers question 1.

I came across a related documentation/discovery problem to what I described for `StyleGuide`, but this time for `BaseReport`.

`help(pep8.BaseReport)` doesn't show all the properties that you would be interested in accessing for a report. For example, if `total_errors` was not mentioned in passing in [this docs](https://github.com/PyCQA/pep8/blob/4dc42d842274ba27d2724e76eb83ff69e7db226f/docs/advanced.rst#automated-tests), I would not know to access it without reading the source.

The fix in this case is to change `BaseReport`'s [declaration](https://github.com/PyCQA/pep8/blob/4dc42d842274ba27d2724e76eb83ff69e7db226f/pep8.py#L1622-L1634):

```python
class BaseReport(object):
    """Collect the results of the checks."""

    print_filename = False
    total_errors = 0
    # Properties here are shown by help().

    def __init__(self, options):
        self._benchmark_keys = options.benchmark_keys
        self._ignore_code = options.ignore_code
        # Results
        self.elapsed = 0
        self.total_errors = 0
        self.counters = dict.fromkeys(self._benchmark_keys, 0)
        self.messages = {}
```

If all the suggestions provided here sound good, I would be happy to open a PR to implement them.

**Comment by nchammas** [comment]
Partly addressed by #500.
